src/NuGet.Clients/NuGet.VisualStudio/Extensibility/IVsPathContextProvider2.cs

```diff
@@ -6,21 +6,35 @@
 namespace NuGet.VisualStudio
 {
     /// <summary>
-    /// A factory to initialize <see cref="IVsPathContext"/> instances.
+    /// A factory to initialize <see cref="IVsPathContext2"/> instances.
     /// </summary>
     [ComImport]
     [Guid("5BAC7095-F674-4778-8788-E15FFF77F96B")]
     public interface IVsPathContextProvider2 : IVsPathContextProvider
     {
         /// <summary>
-        /// Attempts to create an instance of <see cref="IVsPathContext"/> for the solution.
+        /// Attempts to create an instance of <see cref="IVsPathContext2"/> for the solution.
         /// </summary>
         /// <param name="context">The path context associated with this solution.</param>
         /// <returns>
         /// <code>True</code> if operation has succeeded and context was created.
         /// <code>False</code> otherwise.
         /// </returns>
         /// <throws></throws>
-        bool TryCreateSolutionContext(out IVsPathContext context);
+        bool TryCreateSolutionContext(out IVsPathContext2 context);
```

Comment:
Can you add a `remarks` here explaining what @mgoertz-msft pointed out in his comment, so that if in the future anyone wants to avoid the UI thread switch they know the other API will work.

Reply:
it's not just about the UI thread switch since UI thread switch will also irrespective of that. It's about when you already have `solutionDirectory` you've an option to pass it. It's like multiple overloaded options to choose the right one based on your requirements.